Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -797,40 +834,33 @@ lemma map_polynomial : (W.map f).toAffine.polynomial = W.polynomial.map (mapRing
   simp only [polynomial]
   map_simp
 
-lemma map_eval_polynomial (x : R[X]) (y : R) :
-    ((W.map f).toAffine.polynomial.eval <| x.map f).eval (f y) =
-      f ((W.polynomial.eval x).eval y) := by
-  rw [map_polynomial, eval_map, ← coe_mapRingHom, eval₂_hom, coe_mapRingHom, eval_map, eval₂_hom]
+lemma baseChange_polynomial_evalEval_X_Y :
+    (W.baseChange R[X][Y]).toAffine.polynomial.evalEval (C X) Y = W.polynomial := by
+  rw [baseChange, toAffine, map_polynomial, evalEval, eval_map, eval_C_X_eval₂_map_C_X]
```

Comment:
`baseChange_evalEval_polynomial`? Put this together with the other baseChange lemmas?

Reply:
I think `evalEval_ baseChange_polynomial_X_Y` is better ...

Reply:
I'm not sure I should put it with other `baseChange` lemmas, since they're about general base changes but this is about a very specific one. I prefer to make it available immediately after its depency `map_polynomial` is proved.

Reply:
Yeah, fair enough!